src/services/positionService.ts

```diff
@@ -163,8 +163,16 @@ export default class PositionService {
         );
       })
     );
-
-    const results = await Promise.all(promises.map((promise) => promise.catch(() => ({ data: null, error: true }))));
+    const results = await Promise.all(
+      promises.map(async (promise, index) => {
+        const isLastVersion = networksAndVersions[index].version === POSITIONS_VERSIONS[POSITIONS_VERSIONS.length - 1];
+        try {
+          return await timeoutPromise(promise, !isLastVersion ? '30s' : undefined);
+        } catch {
+          return { data: null, error: true };
```

Comment:
should we return a promise too? `Promise.resolve({ data: null, error: true })`

Reply:
I researched the benefits of adding `Promise.resolve()`, and in this case, using `async` ensures the function always returns a promise. However, I can add it for the sake of code consistency.

Reply:
Nah, its gucci as it is!